precompiles/assets/tx.go

```diff
@@ -41,28 +44,33 @@ func (p Precompile) DepositOrWithdraw(
 		return nil, fmt.Errorf(exocmn.ErrContractCaller, err.Error())
 	}
 	// parse the depositTo input params
-	depositWithdrawParams, err := p.DepositWithdrawParamsFromInputs(ctx, args)
+	depositWithdrawParams, err := p.DepositWithdrawParams(ctx, method, args)
 	if err != nil {
 		return nil, err
 	}
 
 	// call assets keeper to perform the deposit or withdraw action
-	switch method.Name {
-	// deposit transactions
-	case MethodDepositTo:
-		depositWithdrawParams.Action = assetstypes.Deposit
-	case MethodWithdraw:
-		depositWithdrawParams.Action = assetstypes.WithdrawPrincipal
-	default:
-		return nil, fmt.Errorf(cmn.ErrUnknownMethod, method.Name)
-	}
 	err = p.assetsKeeper.PerformDepositOrWithdraw(ctx, depositWithdrawParams)
 	if err != nil {
 		return nil, err
 	}
 
+	// call oracle to update the validator info of staker for native asset restaking
+	switch depositWithdrawParams.Action {
+	case assetstypes.DepositNST, assetstypes.WithdrawNST:
+		_, assetID := assetstypes.GetStakerIDAndAssetID(depositWithdrawParams.ClientChainLzID,
+			depositWithdrawParams.StakerAddress, depositWithdrawParams.AssetsAddress)
+		err = p.assetsKeeper.UpdateNativeTokenValidatorListForStaker(ctx, assetID,
```

Comment:
set opAmount with symbol, oracle will know this is a `withdraw` when opAmount is negative

Reply:
Thanks